File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int curr = 0, _max = 0, left = 0, right = height.size() - 1;

        while(left < right){
            curr = (right - left) * min(height[left], height[right]);
            _max = max(_max, curr);
            if(height[left] < height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return _max;
    }
};

File: Data Structures & Algorithms/insert-new-interval/submission-1.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        ans = []

        for s, e in intervals:
            if newInterval == None:
                ans.append([s,e])
                continue
            if s > newInterval[1]:
                if newInterval:
                    ans.append(newInterval)
                    newInterval = None
                    ans.append([s,e])
                    continue
            if e >= newInterval[0]:
                newInterval[0], newInterval[1] = min(newInterval[0], s), max(newInterval[1], e)
            else:
                ans.append([s, e])
        if newInterval:
            ans.append(newInterval)

        print(ans)
        return ans